connection.go

```diff
@@ -52,13 +66,27 @@ func (c *conn) QueryContext(ctx context.Context, query string, args []driver.Nam
 		panic("database/sql/driver: misuse of duckdb driver: QueryContext after Close")
 	}
 
-	if len(args) == 0 {
-		// This should be removed once duckdb_extract_statements and related APIs are available to parse multiple statements
-		// so query cancellation works for this case as well
-		return c.queryUnprepared(query)
+	stmts, size, err := c.extractStmts(query)
+	defer C.duckdb_destroy_extracted(&stmts)
+	if err != nil {
+		return nil, err
 	}
 
-	stmt, err := c.prepareStmt(query)
+	// execute all statements but the last one, apply args only to last statement and return result of last statement
+	for i := C.idx_t(0); i < size-1; i++ {
+		stmt, err := c.prepareExtractedStmt(stmts, i)
+		if err != nil {
+			return nil, err
+		}
+		// send nil args to execute statement and ignore result
+		rows, err := stmt.QueryContext(ctx, nil)
+		rows.Close()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
If the query failed, `rows` will be nil, so this will panic. It should probably close the `stmt` if there was an error, else close the `rows`.

(It's a little ugly that it doesn't support closing the `stmt` in all cases. Might be worth refactoring to support that in a separate PR at some point.)

Reply:
There are actually other cases in this code where the statement is not closed in error cases, such as the return statement in this function. It's also weird that `duckdb_destroy_prepare` is called in one error case, but not others, in `s.execute`.

Do you want to take a stab at fixing these leaks? (I think the solution is to always defer cleanup to `stmt.Close()`, and make it idempotent in case of multiple calls, which I *think* is allowed by the [spec](https://pkg.go.dev/database/sql/driver), though please double-check).

Reply:
I have attempted to address my comment above in https://github.com/marcboeker/go-duckdb/pull/91/commits/ba1759538ddc557470cb90e62c1eff17dcd433a5